File: app/src/main/java/igrek/songbook/domain/song/SongCategoryType.java
package igrek.songbook.domain.song;

public enum SongCategoryType {
	
	ARTIST(1, null),
	
	CUSTOM(2, "song_category_custom"),
	
	OTHERS(3, "song_category_others");
	
	private long id;
	private String localeStringId;
	
	SongCategoryType(long id, String localeStringId) {
		this.id = id;
		this.localeStringId = localeStringId;
	}
	
	public long getId() {
		return id;
	}
	
	public String getLocaleStringId() {
		return localeStringId;
	}
}

File: app/src/main/java/igrek/songbook/domain/song/SongCategoryType.kt
package igrek.songbook.domain.song

import igrek.songbook.R

enum class SongCategoryType(val id: Long, val localeStringId: Int?) {

    ARTIST(1, null),

    CUSTOM(2, R.string.song_category_custom),

    OTHERS(3, R.string.song_category_others)
}
